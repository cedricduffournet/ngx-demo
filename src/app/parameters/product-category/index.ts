export * from './parameter-product-category.component';
