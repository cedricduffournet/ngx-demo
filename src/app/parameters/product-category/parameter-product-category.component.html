<app-product-category-list-view></app-product-category-list-view>
