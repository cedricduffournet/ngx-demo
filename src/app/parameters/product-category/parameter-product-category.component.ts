import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-product-category',
  templateUrl: './parameter-product-category.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterProductCategoryComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
