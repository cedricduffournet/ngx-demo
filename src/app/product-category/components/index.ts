export * from './product-category-add/product-category-add.component';
export * from './product-category-update/product-category-update.component';
export * from './product-category-delete/product-category-delete.component';
export * from './product-category-form/product-category-form.component';
export * from './product-category-item/product-category-item.component';
export * from './product-category-items/product-category-items.component';
