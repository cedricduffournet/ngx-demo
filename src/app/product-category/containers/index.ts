export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
