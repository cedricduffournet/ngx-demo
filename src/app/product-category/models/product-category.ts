import { schema } from 'normalizr';

export interface ProductCategory {
  id: number;
  name: string;
}

export const productCategorySchema = new schema.Entity('productCategories');
