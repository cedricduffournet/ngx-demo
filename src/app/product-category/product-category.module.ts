import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { ProductCategoryEffects } from '@app/product-category/state/effects';
import {
  ProductCategoryAddComponent,
  ProductCategoryUpdateComponent,
  ProductCategoryDeleteComponent,
  ProductCategoryFormComponent,
  ProductCategoryItemComponent,
  ProductCategoryItemsComponent
} from '@app/product-category/components';
import {
  ProductCategoryAddModalComponent,
  ProductCategoryUpdateModalComponent,
  ProductCategoryDeleteModalComponent,
  ProductCategoryListViewComponent
} from '@app/product-category/containers';
import { reducers } from '@app/product-category/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([ProductCategoryEffects]),
    StoreModule.forFeature('productCategories', reducers)
  ],
  declarations: [
    ProductCategoryListViewComponent,
    ProductCategoryAddComponent,
    ProductCategoryUpdateComponent,
    ProductCategoryDeleteComponent,
    ProductCategoryFormComponent,
    ProductCategoryItemComponent,
    ProductCategoryItemsComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryUpdateModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  entryComponents: [
    ProductCategoryUpdateModalComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  exports: [ProductCategoryListViewComponent]
})
export class ProductCategoryModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
