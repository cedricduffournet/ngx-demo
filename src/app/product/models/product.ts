import { schema } from 'normalizr';
import { ProductCategory, productCategorySchema } from '@app/product-category/models/product-category';

export interface Product {
  id: number;
  name: string;
  categories: ProductCategory[];
}

export const productSchema = new schema.Entity('products', {
  categories: [productCategorySchema]
});
