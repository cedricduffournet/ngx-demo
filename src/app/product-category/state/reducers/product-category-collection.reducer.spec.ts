import { reducer } from '@app/product-category/state/reducers/product-category-collection.reducer';
import * as fromProductCategories from '@app/product-category/state/reducers/product-category-collection.reducer';

import {
  ProductCategoryListViewActions,
  ProductCategoryAddModalActions,
  ProductCategoryDeleteModalActions,
  ProductCategoryApiActions
} from '@app/product-category/state/actions';
import { ProductCategory } from '@app/product-category/models/product-category';

describe('ProductCategoryCollectionReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD', () => {
    it('should set loading to true', () => {
      const action = ProductCategoryListViewActions.loadProductCategories();
      const result = reducer(fromProductCategories.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD_SUCCESS ', () => {
    it('should set loading to false, loaded to true and set productCategory ids [1, 2]', () => {
      const initialState = {
        ...fromProductCategories.INITIAL_STATE,
        loading: true
      };

      const productCategories = {
        entities: {
          1: {
            name: 'Name 1'
          },
          2: {
            name: 'Name 2'
          }
        },
        result: [1, 2]
      };

      const action = ProductCategoryApiActions.loadProductCategorySuccess({
        productCategories
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD_FAILURE ', () => {
    const initialState = {
      ...fromProductCategories.INITIAL_STATE,
      loading: true
    };

    it('should set loading to false, loaded to false', () => {
      const error = {
        message: 'error'
      };
      const action = ProductCategoryApiActions.loadProductCategoryFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD', () => {
    it('should set adding to true on ProductCategoryAddModalActions.addProductCategory', () => {
      const productCategory = {
        name: 'AddName'
      } as ProductCategory;
      const action = ProductCategoryAddModalActions.addProductCategory({
        productCategory
      });
      const result = reducer(fromProductCategories.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD_SUCCESS', () => {
    it('should set adding to false, added to true and add new productCategory (3) to ids', () => {
      const initialState = {
        ...fromProductCategories.INITIAL_STATE,
        adding: true,
        ids: [1, 2]
      };
      const productCategory = {
        entities: {
          3: {
            name: 'AddName'
          }
        },
        result: 3
      };
      const action = ProductCategoryApiActions.addProductCategorySuccess({
        productCategory
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD_FAILURE & SHOW_MODAL_ADD', () => {
    const initialState = {
      ...fromProductCategories.INITIAL_STATE,
      adding: true,
      added: true
    };

    it('should set adding to false, added to false', () => {
      const error = {
        message: 'error'
      };
      const action = ProductCategoryApiActions.addProductCategoryFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should set adding to false on ProductCategoryListViewActions.showAddProductCategoryModal', () => {
      const action = ProductCategoryListViewActions.showAddProductCategoryModal();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE', () => {
    it('should set deleting to true', () => {
      const productCategory = {
        id: 1,
        name: 'RemoveName'
      } as ProductCategory;
      const action = ProductCategoryDeleteModalActions.deleteProductCategory({
        productCategory
      });
      const result = reducer(fromProductCategories.INITIAL_STATE, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE_SUCCESS', () => {
    const initialState = {
      ...fromProductCategories.INITIAL_STATE,
      removing: true,
      ids: [1, 2]
    };

    it('should set deleting to false, deleted to true and remove civproductCategoryility (1) form ids', () => {
      const id = 1;

      const action = ProductCategoryApiActions.deleteProductCategorySuccess({
        id
      });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('DELETE_FAILURE & SHOW_MODAL_DELETE', () => {
    const initialState = {
      ...fromProductCategories.INITIAL_STATE,
      removing: true,
      removed: true
    };

    it('should set removing to false, remove to false', () => {
      const error = {
        message: 'error'
      };
      const action = ProductCategoryApiActions.deleteProductCategoryFailure({ error });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should set removing to false, remove to false on ProductCategoryListViewActions.showDeleteProductCategoryModal', () => {
      const action = ProductCategoryListViewActions.showDeleteProductCategoryModal();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    describe('getIds', () => {
      it('should retrieve ids [1,2,3]', () => {
        const result = fromProductCategories.getIds({
          ...fromProductCategories.INITIAL_STATE,
          ids: [1, 2, 3]
        });
        expect(result).toMatchSnapshot();
      });
    });

    describe('getDeleting', () => {
      it('should retrieve removing false', () => {
        const result = fromProductCategories.getDeleting({
          ...fromProductCategories.INITIAL_STATE,
          deleting: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve removing true', () => {
        const result = fromProductCategories.getDeleting({
          ...fromProductCategories.INITIAL_STATE,
          deleting: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getDeleted', () => {
      it('should retrieve deleted false', () => {
        const result = fromProductCategories.getDeleted({
          ...fromProductCategories.INITIAL_STATE,
          deleted: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve deleted true', () => {
        const result = fromProductCategories.getDeleted({
          ...fromProductCategories.INITIAL_STATE,
          deleted: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getAdding', () => {
      it('should retrieve adding false', () => {
        const result = fromProductCategories.getAdding({
          ...fromProductCategories.INITIAL_STATE,
          adding: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve adding true', () => {
        const result = fromProductCategories.getAdding({
          ...fromProductCategories.INITIAL_STATE,
          adding: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getAdded', () => {
      it('should retrieve added false', () => {
        const result = fromProductCategories.getAdded({
          ...fromProductCategories.INITIAL_STATE,
          added: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve added true', () => {
        const result = fromProductCategories.getAdded({
          ...fromProductCategories.INITIAL_STATE,
          added: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getLoaded', () => {
      it('should retrieve loaded false', () => {
        const result = fromProductCategories.getLoaded({
          ...fromProductCategories.INITIAL_STATE,
          loaded: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve loaded true', () => {
        const result = fromProductCategories.getLoaded({
          ...fromProductCategories.INITIAL_STATE,
          loaded: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getLoading', () => {
      it('should retrieve loading false', () => {
        const result = fromProductCategories.getLoading({
          ...fromProductCategories.INITIAL_STATE,
          loading: false
        });

        expect(result).toMatchSnapshot();
      });

      it('should retrieve loading true', () => {
        const result = fromProductCategories.getLoading({
          ...fromProductCategories.INITIAL_STATE,
          loading: true
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
