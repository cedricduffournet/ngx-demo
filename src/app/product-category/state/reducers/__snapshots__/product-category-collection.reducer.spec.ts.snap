// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryCollectionReducer ADD should set adding to true on ProductCategoryAddModalActions.addProductCategory 1`] = `
Object {
  "added": false,
  "adding": true,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer ADD_FAILURE & SHOW_MODAL_ADD should set adding to false on ProductCategoryListViewActions.showAddProductCategoryModal 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer ADD_FAILURE & SHOW_MODAL_ADD should set adding to false, added to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer ADD_SUCCESS should set adding to false, added to true and add new productCategory (3) to ids 1`] = `
Object {
  "added": true,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [
    1,
    2,
    3,
  ],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer DELETE should set deleting to true 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": true,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer DELETE_FAILURE & SHOW_MODAL_DELETE should set removing to false, remove to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "removed": true,
  "removing": true,
}
`;

exports[`ProductCategoryCollectionReducer DELETE_FAILURE & SHOW_MODAL_DELETE should set removing to false, remove to false on ProductCategoryListViewActions.showDeleteProductCategoryModal 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "removed": true,
  "removing": true,
}
`;

exports[`ProductCategoryCollectionReducer DELETE_SUCCESS should set deleting to false, deleted to true and remove civproductCategoryility (1) form ids 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": true,
  "deleting": false,
  "ids": Array [
    2,
  ],
  "loaded": false,
  "loading": false,
  "removing": true,
}
`;

exports[`ProductCategoryCollectionReducer LOAD should set loading to true 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": true,
}
`;

exports[`ProductCategoryCollectionReducer LOAD_FAILURE  should set loading to false, loaded to false 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer LOAD_SUCCESS  should set loading to false, loaded to true and set productCategory ids [1, 2] 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [
    1,
    2,
  ],
  "loaded": true,
  "loading": false,
}
`;

exports[`ProductCategoryCollectionReducer Selectors getAdded should retrieve added false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getAdded should retrieve added true 1`] = `true`;

exports[`ProductCategoryCollectionReducer Selectors getAdding should retrieve adding false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getAdding should retrieve adding true 1`] = `true`;

exports[`ProductCategoryCollectionReducer Selectors getDeleted should retrieve deleted false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getDeleted should retrieve deleted true 1`] = `true`;

exports[`ProductCategoryCollectionReducer Selectors getDeleting should retrieve removing false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getDeleting should retrieve removing true 1`] = `true`;

exports[`ProductCategoryCollectionReducer Selectors getIds should retrieve ids [1,2,3] 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`ProductCategoryCollectionReducer Selectors getLoaded should retrieve loaded false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getLoaded should retrieve loaded true 1`] = `true`;

exports[`ProductCategoryCollectionReducer Selectors getLoading should retrieve loading false 1`] = `false`;

exports[`ProductCategoryCollectionReducer Selectors getLoading should retrieve loading true 1`] = `true`;

exports[`ProductCategoryCollectionReducer undefined action should return the default state 1`] = `
Object {
  "added": false,
  "adding": false,
  "deleted": false,
  "deleting": false,
  "ids": Array [],
  "loaded": false,
  "loading": false,
}
`;
